src/data/dfinsupp/interval.lean

```diff
@@ -167,4 +167,21 @@ lemma card_Ioo : (Ioo f g).card = ∏ i in f.support ∪ g.support, (Icc (f i) (
 by rw [card_Ioo_eq_card_Icc_sub_two, card_Icc]
 
 end locally_finite
+
+section canonically_ordered
+variables [decidable_eq ι] [Π i, decidable_eq (α i)]
+variables [Π i, canonically_ordered_add_monoid (α i)] [Π i, locally_finite_order (α i)]
+
+variables (f : Π₀ i, α i)
+
+lemma card_Iic : (Iic f).card = ∏ i in f.support, (Iic (f i)).card :=
+by simp_rw [Iic_eq_Icc, card_Icc, dfinsupp.bot_eq_zero, support_zero, empty_union, zero_apply,
+  bot_eq_zero]
+
+lemma card_Iio : (Iio f).card = ∏ i in f.support, (Iic (f i)).card - 1 :=
+by simp_rw [Iio_eq_Ico, card_Ico, dfinsupp.bot_eq_zero, support_zero, empty_union, zero_apply,
+  Iic_eq_Icc, bot_eq_zero]
```

Comment:
Use (the variant of) `finset.card_Ico_eq_card_Icc_sub_one`.

Reply:
I think I added that lemma after writing this one, good catch.